packages/std/src/api.rs

```diff
@@ -0,0 +1,349 @@
+/// This maintains types needed for a public API
+/// In particular managing serializing and deserializing errors through API boundaries
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use crate::errors::{Error, SystemError};
+use crate::HumanAddr;
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "lowercase")]
+pub enum ApiResult<T, E: std::error::Error = ApiError> {
+    Ok(T),
+    Err(E),
+}
+
+impl<T: Into<U>, U, E: std::error::Error> Into<Result<U, E>> for ApiResult<T, E> {
+    fn into(self) -> Result<U, E> {
+        match self {
+            ApiResult::Ok(t) => Ok(t.into()),
+            ApiResult::Err(e) => Err(e),
+        }
+    }
+}
```

Comment:
What is this conversion needed for? Isn't it giving the false impression that `ApiError` and `Error` can be mapped both ways?

Reply:
I want to recover normal `Result` types, so all the functionality associated with them is exposed, not just `unwrap()` and `?`, but `map`, `context`, ...

`ApiResult<T, ApiError> => Result<T, ApiError>` just fine. You just can't "upgrade" the error part. This should be defined in the fact that both objects have the same error types.